gui/tools/autorepl.py

```diff
@@ -314,12 +314,13 @@ def write_results():
     sshproc = pipeopen('%s %s' % (sshcmd, rzfscmd))
     output, error = sshproc.communicate()
     remote_zfslist = {}
-    for i in re.sub(r'[ \t]+', ' ', output, flags=re.M).splitlines():
-        data = i.split()
+    for i in re.sub(r'\t+', ' ', output, flags=re.M).splitlines():
+        readonly_value = i.split()[-1]
+        data = [i.split(readonly_value)[0].rstrip(), readonly_value]
         remote_zfslist[data[0]] = {'readonly': data[1] == 'on'}
 
     # Attempt to create the remote dataset.  If it fails, we don't care at this point.
-    rzfscmd = "zfs create -o readonly=on "
+    rzfscmd = "\'" + "zfs create -o readonly=on "
```

Comment:
Do you want `"'zfs create -o readonly=on "`?

Reply:
Correct, the zfs command needs to be prefixed with a single-quote so that I can escape the zpool/dataset with double-quotes on line 340.

Reply:
Why so cryptic then?

Reply:
I'm not sure I understand? Would you like me to add the single-quote inside the double-quote instead of using the "+" operator?

Reply:
`rzfscmd = "'zfs create -o readonly=on "`

Reply:
Okay, done.